docs/reference/config-params.md

```diff
@@ -805,124 +932,157 @@ Again, it is ideal for exposing a service or app to the internet under a single
 
+## x-k8s.service.nodeport
 
-## kev.service.nodeport.port
-
-Defines type Node Port value for Kubernetes service of `NodePort` type. See official K8s [documentation](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport).
+Defines the Node Port value for a Kubernetes service of type `NodePort`. See official K8s [documentation](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport).
 NOTE: `nodeport` attributes will be ignored for any other service type!
 Kev will attempt to extract that information from the compose configuration.
 
 ### Default: `nil` - no nodeport defined by default!
 
 ### Possible options: Arbitrary integer. Example `10222`.
 
-> kev.service.nodeport.port:
+> x-k8s.service.nodeport:
 ```yaml
 version: 3.7
 services:
   my-service:
-    labels:
-      kev.service.nodeport.port: 5555
+    x-k8s:
+      service:
+        type: nodeport
+        nodeport: 5555
 ...
 ```
 
-## kev.service.expose
+## x-k8s.service.expose
 
-Defines whether to expose the service to external world. This detail can't be easily derived from the compose file and so in order to expose a service to external world user must explicitly instruct Kev to do so. By default all component services aren't exposed i.e. have no ingress attached to them.
+Defines how to expose the service externally. This detail can't be easily derived from the compose file and so in order to expose a service the user must explicitly instruct Kev to do so. By default, all component services aren't exposed i.e. have no ingress attached to them.
 
-### Default: `""` - No ingress will be created!
+### x-k8s.service.expose.domain
 
-### Possible options:
-* `"true"` - ingress will be created with Kubernetes cluster defaults
-* `"domain.com,otherdomain.com..."` - comma separated list of domains for the ingress.
+This setting accepts a comma separated list of domains for the ingress.
 
-> kev.service.expose:
+E.g. `"domain.com,otherdomain.com..."`
+
+#### Default: `""` - No ingress will be created!
+
+> x-k8s.service.expose.domain:
 ```yaml
 version: 3.7
 services:
   my-service:
-    labels:
-      kev.service.expose: "my-awesome-service.com"
+    x-k8s:
+      service:
+        type: LoadBalancer
+        expose:
+          domain: "my-awesome-service.com"
 ...
 ```
 
-## kev.service.expose.tls-secret
+### x-k8s.service.expose.tlsSecret
 
 Defines whether to use TLS for the exposed service and which secret name contains certificates for the service. See official K8s [documentation](https://kubernetes.io/docs/concepts/services-networking/ingress/#tls).
 
-NOTE: This option is only relevant when service is exposed, see: [kev.service.expose](#kev-service-expose) above.
+NOTE: This option is only relevant when service is exposed, see: [x-k8s.service.expose.domain](#x-k8s.service.expose.domain) above.
 
-### Default: `nil` - No TLS secret name specified by default!
+#### Default: `""` - No TLS secret name specified by default!
```

Comment:
Have we changes the way it's handled internally now? For things that can practically be defined at all maybe we should allow nil value and default to nil? I understand that the config struct now defines that as string hence this change?

Reply:
Yup, internally, it's a string.

Will revert back to `nil` ... this should work with strings too.